### PR TITLE
Tag Directory Page

### PR DESCRIPTION
**Pull request by njd9419**
This PR adds a Tag Directory Page to the site. It's pretty minimal right now, but will probably look a bit nicer once we get some feedback from Dr. Decker.

Perhaps adding filtering to this page may be a good idea in the future.

Screenshot:
image

### CODE REVIEW

File: client/src/actions/tagDirectory.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { FETCH_TAGS_PENDING, FETCH_TAGS_SUCCESS, FETCH_TAGS_ERROR } from '../constants';
```
Comment (benvogler) [inline client/src/actions/tagDirectory.js]: Is it a good idea to reuse the constants? doesn't this mean that the reducer will affect the other places we use these constants?

File: client/src/components/TagColumn/TagColumn.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import './TagColumn.scss';
+import Tags from '../Tags/Tags';
+
+/**
+ * Presentational component for a column of Tags
+ */
+export default class TagColumn extends React.Component {
```
Comment (benvogler) [inline client/src/components/TagColumn/TagColumn.js]: Why is this component necessary?

File: client/src/pages/TagDirectoryPage/TagDirectoryPage.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    static mapStateToProps(state) {
+        return state.tagDirectoryPage;
+    }
+
+    render() {
+        const { tags, fetchTagsPending } = this.props;
+
+        const allTags = [];
```
Comment (benvogler) [inline client/src/pages/TagDirectoryPage/TagDirectoryPage.js]:
```suggestion
        const allTags = tags.filter(tag => !tag.isMaterial);
```

File: client/src/pages/TagDirectoryPage/TagDirectoryPage.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        const allMaterials = [];
```
Comment (benvogler) [inline client/src/pages/TagDirectoryPage/TagDirectoryPage.js]:
```suggestion
        const allMaterials = tags.filter(tag => tag.isMaterial);
```

File: client/src/actions/tagDirectory.js (hunk 1 above)
Comment (njd9419) [reply client/src/actions/tagDirectory.js]: They are not being re-used. I added them to `constants.js` down below.

Comment (njd9419) [reply client/src/actions/tagDirectory.js]: I can make them a little more specific to the Tag Directory Page though if you think that's needed.

File: client/src/components/TagColumn/TagColumn.js (hunk 2 above)
Comment (njd9419) [reply client/src/components/TagColumn/TagColumn.js]: When I first started I thought it'd be more, but it's not. I'll remove it.

File: client/src/actions/tagDirectory.js (hunk 1 above)
Comment (benvogler) [reply client/src/actions/tagDirectory.js]: Ah my bad. Maybe they should be more specific but it's not super important

**Pull request merged**